pkgs/development/beam-modules/fetch-mix-deps.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenvNoCC, elixir, hex, rebar, rebar3, cacert }:
+
+{ name, version, sha256, src, mixEnv ? "prod", debug ? false, meta ? { } }:
+
+with stdenvNoCC.lib;
+
+stdenvNoCC.mkDerivation ({
+  name = "mix-deps-${name}-${version}";
+
+  phases = [ "configurePhase" "downloadPhase" ];
+
+  nativeBuildInputs = [ elixir hex cacert ];
+
+  inherit src;
+
+  MIX_ENV = mixEnv;
+  MIX_REBAR = "${rebar}/bin/rebar";
+  MIX_REBAR3 = "${rebar3}/bin/rebar3";
+  MIX_DEBUG = if debug then 1 else 0;
+
+  configurePhase = ''
+    mkdir -p $out/deps
+    mkdir -p $out/.hex
```

Comment:
Is there a need to export the internal hex directory? Doesn't this make the derivation non-deterministic?

Reply:
Since the .hex contains all the tarballs, I was thinking it could be re-used for caching. I have to admit I'm not sure how to do that.
The idea being that if we that gets deleted, then on a new build, we need to re-download all the dependencies again.

Reply:
So far I haven't found a way to re-use that directory. Unless we find how to re-use that, the default should be to not export it.

Reply:
I ended up removing the export of the .hex folder.
Here is what I tried.
- giving a .hex folder as an input so that the downloaded files could be used as cache. The problem with that is that if you just give a path name, the output would not be fixed. What I ended up thinking was that the caching step should be taken care of when the process of getting and storing dependency is going through the nix store.